lib/neo4j/active_node/query/query_proxy_methods.rb

```diff
@@ -94,12 +94,13 @@ def include?(other, target = nil)
         end
 
         def exists?(node_condition = nil, target = nil)
-          unless node_condition.is_a?(Integer) || node_condition.is_a?(Hash) || node_condition.nil?
-            fail(Neo4j::InvalidParameterError, ':exists? only accepts neo_ids')
+          unless [Integer, Fixnum, String, Hash, NilClass].include?(node_condition.class)
+            fail(Neo4j::InvalidParameterError, ':exists? only accepts ids or conditions')
           end
           query_with_target(target) do |var|
             start_q = exists_query_start(node_condition, var)
-            start_q.query.reorder.return("COUNT(#{var}) AS count").first.count > 0
+            result = start_q.query.reorder.return("ID(#{var}) AS found_ids LIMIT 1").first
+            !!result && result.found_ids > 0
```

Comment:
Shouldn't `found_ids` be `found_id`?  We're returning one row with one column and then I think we should be checking to see if the size of `result` is > 0, right?